tests/bsim/bluetooth/mesh/overlay_pst.conf

```diff
@@ -4,6 +4,7 @@ CONFIG_FLASH=y
 CONFIG_FLASH_MAP=y
 CONFIG_NVS=y
 CONFIG_SECURE_STORAGE=y
+CONFIG_SECURE_STORAGE_ITS_IMPLEMENTATION_ZEPHYR=y
```

Comment:
Seems my PR(https://github.com/zephyrproject-rtos/zephyr/pull/84237) sneaked before yours. You need to put this option into another overlay instead: https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/bluetooth/mesh/overlay_ss.conf otherwise after rebasing mesh bsim will be broken.